### PR TITLE
Use device registry for casts and spotify devices

### PR DESCRIPTION
**Pull request by fondberg**
This PR replaces #176. #177 got added as well but it should not break anything (tired and forgot to rebase)

### CODE REVIEW

File: custom_components/spotcast/sensor.py [hunk 1]
```diff
@@ -42,29 +40,26 @@ def device_state_attributes(self):
         """Return the state attributes."""
         return self._attributes
 
+    # ChromecastInfo(services={ServiceInfo(type='mdns', data='Google-Home-fa4dda8024b636eb450ca54f021a264a._googlecast._tcp.local.')}, uuid='fa4dda80-24b6-36eb-450c-a54f021a264a', _manufacturer=None, model_name='Google Home', friendly_name='Kök', is_audio_group=False, is_dynamic_group=None),
```
Comment (KapJI) [inline custom_components/spotcast/sensor.py:43]: Leftover from debugging?

File: custom_components/spotcast/__init__.py [hunk 2]
```diff
@@ -10,10 +10,12 @@
...
 from homeassistant.exceptions import HomeAssistantError
 import homeassistant.helpers.config_validation as cv
-from homeassistant.components.cast.media_player import KNOWN_CHROMECAST_INFO_KEY
+from homeassistant.components.cast.media_player import CastDevice
 from homeassistant.components.cast.helpers import ChromeCastZeroconf
+from homeassistant.components.spotify.media_player import SpotifyMediaPlayer
+from homeassistant.helpers import entity_platform
 
-__VERSION__ = "3.4.7"
+__VERSION__ = "3.5.2"
```
Comment (KapJI) [inline custom_components/spotcast/__init__.py:18]: It's now inconsistent with `manifest.json`. I think it's better to drop it and keep version only in `manifest.json`.

**Review by KapJI — APPROVED** [review]
LGTM

File: custom_components/spotcast/sensor.py (hunk 1 above)
Comment (fondberg) [reply custom_components/spotcast/sensor.py]: yup

**Comment by hmmbob** [comment]
@fondberg It works for me, yet it does throw errors in the log. But - it works in my use case.

Error:
```
2021-04-11 19:22:02 ERROR (SyncWorker_14) [custom_components.spotcast] No device with id "bf7485e4bdc069dfd996748826af2f02" known by Spotify
2021-04-11 19:22:02 ERROR (SyncWorker_14) [custom_components.spotcast] Known devices: []
2021-04-11 19:23:20 ERROR (SyncWorker_11) [custom_components.spotcast] No device with id "bf7485e4bdc069dfd996748826af2f02" known by Spotify
2021-04-11 19:23:20 ERROR (SyncWorker_11) [custom_components.spotcast] Known devices: []
```

Used HA script:

```yaml
#############################
## Start sleeping music     #
#############################
start_music_t:
  alias: Slaaptijd
  mode: restart
  sequence:
    - service: spotcast.start
      data:
        account: "t"
        entity_id: media_player.hmt
        uri: "spotify:track:5OoSxUZSbP1p7kb1d0SozG"
```

**Pull request merged**